core/dbt/include/global_project/macros/materializations/tests/test.sql

```diff
@@ -6,15 +6,17 @@
 
     {% set identifier = model['alias'] %}
     {% set old_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) %}
+
+    {% set relation_type = config.get('store_failures_as', 'table') %}
     {% set target_relation = api.Relation.create(
-        identifier=identifier, schema=schema, database=database, type='table') -%} %}
+        identifier=identifier, schema=schema, database=database, type=relation_type) -%} %}
 
     {% if old_relation %}
         {% do adapter.drop_relation(old_relation) %}
     {% endif %}
 
     {% call statement(auto_begin=True) %}
-        {{ create_table_as(False, target_relation, sql) }}
+        {{ get_create_sql(target_relation, sql) }}
```

Comment:
It's worth noting this potentially provides shadow support for materialized views with default configurations. This is not tested, and is left as a future.

Reply:
This is no longer the case as we raise an exception for unsupported values of `store_failures_as`.